realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -441,33 +441,45 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
 
                 // make sure that nullability matches
                 if (metadata.isNullable(field)) {
-                    writer.beginControlFlow("if (!table.isColumnNullable(%s))", fieldIndexVariableReference(field));
-                    if (Utils.isBoxedType(fieldTypeCanonicalName)) {
+                    // Check if the existing PrimaryKey does support null value for String, Byte, Short, Integer, & Long
+                    if (field.equals(metadata.getPrimaryKey()) && Utils.isNullablePrimaryKeyType(field)) {
```

Comment:
I think I need feedback on this. It is in fact _only_ necessary to see if a type is nullable. But then we have benefit of generating detailed migration error message about the breaking change to users. 

It could be removed in a minute if deemed unnecessary.